contracts/actions/fee/GasFeeTaker.sol

```diff
@@ -26,53 +26,56 @@ contract GasFeeTaker is ActionBase, GasFeeHelper {
         inputData.dfsFeeDivider = _parseParamUint(inputData.dfsFeeDivider, _paramMapping[2], _subData, _returnValues);
 
-        uint256 txCost = calcGasCost(inputData.gasUsed, inputData.feeToken, 0);
-
         /// @dev This means inputData.availableAmount is not being piped into
         /// @dev To stop sender from sending any value here, if not piped take proxy balance
         if (_paramMapping[1] == 0) {
             inputData.availableAmount = inputData.feeToken.getBalance(address(this));
         }
 
+        uint256 amountLeft = _takeFee(inputData);
+
+        emit ActionEvent("GasFeeTaker", abi.encode(inputData, amountLeft));
+        return bytes32(amountLeft);
+    }
+
+    function _takeFee(GasFeeTakerParams memory _inputData) internal returns (uint256 amountLeft) {
+        uint256 txCost = calcGasCost(_inputData.gasUsed, _inputData.feeToken, 0);
+
         // cap at 20% of the max amount
-        if (txCost >= (inputData.availableAmount / 5)) {
-            txCost = inputData.availableAmount / 5;
+        if (txCost >= (_inputData.availableAmount / 5)) {
+            txCost = _inputData.availableAmount / 5;
         }
 
-        if (inputData.dfsFeeDivider != 0) {
+        if (_inputData.dfsFeeDivider != 0) {
             /// @dev If divider is lower the fee is greater, should be max 5 bps
-            if (inputData.dfsFeeDivider < MAX_DFS_FEE) {
-                inputData.dfsFeeDivider = MAX_DFS_FEE;
+            if (_inputData.dfsFeeDivider < MAX_DFS_FEE) {
+                _inputData.dfsFeeDivider = MAX_DFS_FEE;
             }
 
             // add amount we take for dfs fee as well
-            txCost += inputData.availableAmount / inputData.dfsFeeDivider;
+            txCost += _inputData.availableAmount / _inputData.dfsFeeDivider;
         }
 
-        uint256 amountLeft = sub(inputData.availableAmount, txCost);
-
-        inputData.feeToken.withdrawTokens(feeRecipient.getFeeAddr(), txCost);
-
-        emit ActionEvent("GasFeeTaker", abi.encode(inputData, amountLeft));
-        return bytes32(amountLeft);
+        amountLeft = sub(_inputData.availableAmount, txCost);
+        _inputData.feeToken.withdrawTokens(feeRecipient.getFeeAddr(), txCost);
     }
 
     /// @inheritdoc ActionBase
     // solhint-disable-next-line no-empty-blocks
-    function executeActionDirect(bytes memory _callData) public payable override {}
+    function executeActionDirect(bytes memory _callData) public payable virtual override {}
 
     /// @inheritdoc ActionBase
     function actionType() public pure virtual override returns (uint8) {
         return uint8(ActionType.FEE_ACTION);
     }
 
-    function parseInputs(bytes memory _callData) public pure returns (Params memory inputData) {
-        inputData = abi.decode(_callData, (Params));
+    function parseInputsGasFeeTaker(bytes memory _callData) public pure returns (GasFeeTakerParams memory inputData) {
```

Comment:
any specific reason for changing function and struct names
Full name of the struct would go from GasFeeTaker.Params to GasFeeTaker.GasFeeTakerParams and parseInputs is called that way in every action, should we change it in others, or revert this change here for consistency?